code/parachain/frame/pallet-multihop-xcm-ibc/src/lib.rs

```diff
@@ -455,17 +457,19 @@ pub mod pallet {
 
 			let (pallet_id, address_from, route_id, mut addresses) = location_info;
 
+			/// what if PalletInstanceId is const, just asking, can you match on it?
 			if *pallet_id != T::PalletInstanceId::get() {
 				<Pallet<T>>::deposit_event(crate::Event::<T>::FailedCallback {
 					origin_address: address_from,
 					route_id,
-					reason: 1,
+					reason: 1, // what is reason? 
```

Comment:
Just a number to help for front end understand why it failed when trace events.
Will replace with enum and will return corresponding error

Reply:
yeah, please error. i dislike using events like that so. you could return error and issue one error event in one place. 

but okey with events, but events cannot be ? with https://doc.rust-lang.org/std/ops/trait.FromResidual.html , so code is really hard to read